master/pkg/tasks/task.go

```diff
@@ -261,7 +287,10 @@ func injectUserArchive(aug *model.AgentUserGroup, workDir string) cproto.RunArch
 	passwdBytes := []byte(
 		fmt.Sprintf("%v:x:%v:%v::%v:/bin/bash\n", aug.User, aug.UID, aug.GID, workDir),
 	)
-	shadowBytes := []byte(fmt.Sprintf("%v:!!:::::::\n", aug.User))
+	// Add a bogus password in the shadow file.  Cannot use ! as that locks the account
+	// when using SLURM/Singularity.   Use * which cannot match any password.
+	// Our sshd_config is configured with "PasswordAuthentication no" so * works.
```

Comment:
is this line still necessary?  Or is it from the old version?

I don't understand why PasswordAuthentication makes a difference as to whether or not `*` works.

Reply:
Updated the comment.